src/main/java/edu/kis/powp/jobs2d/command/ICompoundCommand.java

```diff
@@ -8,7 +9,15 @@
  */
 public interface ICompoundCommand extends DriverCommand {
 
-	public Iterator<DriverCommand> iterator();
+	Iterator<DriverCommand> iterator();
 
-	public void setCommands(List<DriverCommand> collect);
+	@Override
+	default DriverCommand clone() throws CloneNotSupportedException {
+		List<DriverCommand> commands = new ArrayList<>();
+		Iterator<DriverCommand> iter = iterator();
+		while (iter.hasNext()) {
+			commands.add(iter.next().clone());
+		}
```

Comment:
If you could add Iterable interface to this interface then we could use foreach loop, more readable:
for (DriverCommand command: this)
Right?

Reply:
Done in next commit